### ISSUE TITLE
Is it possible to perform hyperparameter tuning in bertopic incremental model? (using the River package with DBSTREAM) 

### ISSUE THREAD

**Issue by jjaitley** [MaartenGr/BERTopic#1071]
Great package @MaartenGr awesome work done. I have a query. My use case involves making new clusters if something new pops up in my data. The default model trained using the river package as suggested in the documentation is working fine for me. I was trying to come up with the most optimized parameters for my topic model focusing on Quality (density or separation) of clusters & Distribution of topics. Since online TPM doesn't support HDBSCAN, what are my options to calculate something like npmi or silhouette_score. It will be great if an example can be shared using the river package which uses DBSTREAM to calculate npmi or silhouette_score

**Comment by MaartenGr** [comment]
You would have to calculate those metrics yourself. Here, it might be important to check when you are calculating those metrics. Since you have an online use case, it might not make sense to only calculate those metrics at the end of the training loop but also in between. A sort of loss curve that you would have when training a neural net. Moreover, depending on the use case, it might also be worthwhile to introduce some sort of decay if you are interested in local temporal performance.
